Update link to our Privacy Policy history in our Privacy Policy (#263)
OpenTermsArchive/opentermsarchive.org: content/pages/en/privacy-policy.mdx

> <p className="text__smallcaps mb__3XL">Last updated: September 26, 2023</p>

Reviewer: So, should we update the “last updated” date for such a minor change or not? 😉

Author: Yep we should

Author: "Last update" date is not related to the importance of the changes for me.

> This privacy policy explains how, why and under what conditions the Open Terms Archive (OTA) site collects personal information and how it is used. Our privacy policy will change over time. And of course, we also record the changes of [our own privacy policy](https://github.com/OpenTermsArchive/demo-versions/commits/main/Open%20Terms%20Archive/Privacy%20Policy.md) 😉

Reviewer: ```suggestion
This privacy policy explains how, why and under what conditions the Open Terms Archive (OTA) site collects personal information and how it is used. Our privacy policy will change over time. And of course, we [publicly record those changes](https://github.com/OpenTermsArchive/demo-versions/commits/main/Open%20Terms%20Archive/Privacy%20Policy.md) with Open Terms Archive 😉
```
